src/main/scala/com/horizon/exchangeapi/ExchangeApiApp.scala

```diff
@@ -307,51 +336,113 @@ object ExchangeApiApp extends App with OrgsRoutes with UsersRoutes with NodesRou
       case _ => logger.debug("invalid case sent to MsgsCleanupActor")
     }
   }
-  val msgsCleanupActor = system.actorOf(Props(classOf[MsgsCleanupActor]))
-  var msgsCleanup : Cancellable = _
-  val msgsCleanupInterval = ExchConfig.getInt("api.defaults.msgs.expired_msgs_removal_interval")
+  val msgsCleanupActor: ActorRef = system.actorOf(Props(classOf[MsgsCleanupActor]))
+  var msgsCleanup: Cancellable = _
+  val msgsCleanupInterval: Int = ExchConfig.getInt("api.defaults.msgs.expired_msgs_removal_interval")
   logger.info("Remove expired msgs cleanup Interval: " + msgsCleanupInterval.toString)
-
-
-  // Start serving client requests
-  val serverBinding: Future[Http.ServerBinding] = Http().bindAndHandle(routes, ExchangeApi.serviceHost, ExchangeApi.servicePort)
-
-  // Configure graceful termination. See: https://doc.akka.io/docs/akka-http/current/server-side/graceful-termination.html
-  // But also see: https://discuss.lightbend.com/t/graceful-termination-on-sigterm-using-akka-http/1619
-  // Note: can't test this in sbt. Instead use 'make runexecutable' and then ctrl-c
-  val secondsToWait = ExchConfig.getInt("api.service.shutdownWaitForRequestsToComplete")  // ExchConfig.getAkkaConfig() also makes the akka unbind phase this long
-  CoordinatedShutdown(system).addTask(CoordinatedShutdown.PhaseServiceUnbind, "http_shutdown") { () =>
-    serverBinding.flatMap({ s =>
-      println(s"Exchange server unbound, waiting up to $secondsToWait seconds for in-flight requests to complete...")
+  
+  val secondsToWait: Int = ExchConfig.getInt("api.service.shutdownWaitForRequestsToComplete") // ExchConfig.getAkkaConfig() also makes the akka unbind phase this long
+  
+  var serverBindingHttp: Future[Http.ServerBinding] = null
+  var serverBindingHttps: Future[Http.ServerBinding] = null
+  
+  if(ExchangeApi.servicePortEncrypted.isDefined) {
```

Comment:
In the other mgmt hub components, the decision of whether to listen on https is based on whether the key and cert have been mounted into the container. This approach has 2 advantages:

1. You can default the encrypted port number
2. You avoid the user error that they specified an encrypted port number, but not the key